xy/area_centroid.go

```diff
@@ -155,9 +155,9 @@ func (calc *AreaCentroidCalculator) addTriangle(p0, p1, p2 geom.Coord, isPositiv
 	}
 	centroid3(p0, p1, p2, calc.triangleCent3)
 	area2 := area2(p0, p1, p2)
-	calc.cg3[0] += sign * area2 * calc.triangleCent3[0]
-	calc.cg3[1] += sign * area2 * calc.triangleCent3[1]
-	calc.areasum2 += sign * area2
+	calc.cg3[0] += float64(sign * area2 * calc.triangleCent3[0])
```

Comment:
All the variables here (`sign`, `area2`, and `calc.triangleCent3[0]`) are already `float64`s, so what effect does the cast to `float64` have?

Same question for the rest of the changes in this function.